### DISCUSSION TITLE
web range input

### DISCUSSION THREAD

**Discussion by miaoudak** [arendst/Tasmota#17591] [Q&A]
I have created a driver with Berry that adds a slider in the main page and I want when the slider changes to pass its value in a global varable. Is this possible?

**Comment by sfromis** [comment]
If you have Berry code activated when the slider is moved, it should be very easy to set a global variable in the Berry environment. Not seeing your code, I'm not sure if there just might be hurdles for this....

**Reply by miaoudak** [reply]
first of all thank you for your reply.
well I have the following code.

`import webserver 
var test=50

def setTest (cmd, idx, payload, payload_json)
    test=number(payload)
  end
tasmota.add_cmd("setTest",setTest)

class MyButtonMethods : Driver
  def web_add_main_button()
    webserver.content_send("<p></p><button style=\"background:#bb0000;\" color: #1e1e1e;\" onclick='la(\"&test_Clicked=1\");'>test</button>") 
  end

  def web_sensor()
    webserver.content_send("<div style=\"width:100%;text-align:center;font-weight:normal;font-size:62px\">" ..test.." </div>")
    webserver.content_send("<input  type=\"range\" min=\"0\" max=\"20\" value=\""..test .."\" oninput=\" 'la(\"&m_change_Target_Temp=1\" >")

    if webserver.has_arg("test_Clicked")
      print("test pressed")
    end

  end
end
d2 = MyButtonMethods()
tasmota.add_driver(d2)`

which produces the ui:
![image](https://user-images.githubusercontent.com/12726427/210568034-f4f3b955-50dc-4414-99ed-4a01fee25a4f.png)

I have added a command that changes the test value and when using it the slider is updated in the UI. 
how I can go the other way arround that is when I drag the slider, the test value to bes updated???

**Reply by sfromis** [reply]
Seen more narrowly, if you want to have the value of the slider in the Berry code, you should include it in the request you send from the JavaScript code in the slider, `la("&m_change_Target_Temp=1"` - as it is now, you are only sending constant 1. If you instead send the value, I think (not tested this) that the Berry code would be able to access it this way, and update your global variable.
```
if webserver.has_arg("m_change_Target_Temp")
  var position = webserver.arg("m_change_Target_Temp")
  print("test pressed", position)
  test = position
end
```
